File: bank-account-project/src/main/java/com/example/bank_account_project/dto/UserDto.java
package com.example.bank_account_project.dto;

import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserDto {

    private Long userId;

    private String userName;

    private String userFirstName;

    private String userIdentity;

    private String userEmail;

    private String userPhoneNumber;

    private String userAddress;

    private String userJobTitle;

    private String userDateOfBirth;

    private Double userAccountBalance;

}

File: web-app-angular/src/app/models/user.dto.ts
export class UserDto {
  userId: number | null = null;
  userName: string | null = null;
  userFirstName: string | null = null;
  userEmail: string | null = null;
  userPhoneNumber: string | null = null;
  userAddress: string | null = null;
  userJobTitle: string | null = null;
  userDateOfBirth: string | null = null;
  userAccountBalance: string | null = null;
  userIdentity: string | null = null;

  // Adding an index signature to allow dynamic property access
  [key: string]: string | number | null;

  constructor(init?: Partial<UserDto>) {
    Object.assign(this, init);
  }
}
